pandas/tests/extension/base/groupby.py

```diff
@@ -108,9 +108,13 @@ def test_groupby_extension_transform(self, data_for_grouping):
 
     def test_groupby_extension_apply(self, data_for_grouping, groupby_apply_op):
         df = pd.DataFrame({"A": [1, 1, 2, 2, 3, 3, 1, 4], "B": data_for_grouping})
-        df.groupby("B", group_keys=False).apply(groupby_apply_op)
+        msg = "DataFrameGroupBy.apply operated on the grouping columns"
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            df.groupby("B", group_keys=False).apply(groupby_apply_op)
```

Comment:
In #54915 I met a similar problem having to rewrite a ton of tests to account for a deprecation message. I preferred doing a function to avoid having to type out the error message so often.

My preference for these tests would be to either tests using a fixture `include_groups` with True/False options, or to test using `include_groups=False`. I would not test using `include_groups=True` (old behavior) only.

 I would be ok to test using `include_groups=False` by default and go with `include_groups=True|False` where appropriate (testing fallthrough in the try/except etc.).

EDIT: Don't know the general attritute if others want to both `include_groups=True|False` for all tests.

Reply:
> I preferred doing a function to avoid having to type out the error message so often.

I'll keep this in mind for the future, but the PR is already setup and so I'd prefer not making changes at this point unless they add a benefit.

The benefit of testing with `include_groups=True` is that when we enforce the deprecation, there isn't a risk of missing any code to cleanup.

Reply:
Ok, no problem. I haven't received feedback on #54915, so actually not confirmed if others agree with that approach.